roles/ceph-update/tasks/main.yml

```diff
@@ -0,0 +1,48 @@
+---
+# Wait for a max of 30 minutes
+- name: wait for ceph cluster to be in health ok state
+  shell: ceph health | egrep -q "HEALTH_OK"
+  register: result
+  until: result.rc == 0
+  retries: 30
+  delay: 60
+  delegate_to: "{{ groups['ceph_monitors'][0] }}"
+
+- name: restart ceph mons
+  service: name=ceph-mon-all state=restarted
+
+# Setting noout to prevent cluster from rebalancing after service restart
+- name: set osd noout
+  command: ceph osd set noout
+  delegate_to: "{{ groups['ceph_monitors'][0] }}"
+  when: inventory_hostname == groups['ceph_osds'][0]
+
+# Setting osd values for first osd to reduce performance degradation of client I/O
+- name: set osd noscrub
+  command: ceph osd set noscrub
+  delegate_to: "{{ groups['ceph_monitors'][0] }}"
+  when: inventory_hostname == groups['ceph_osds'][0]
+
+- name: set osd nodeep scrub
+  command: ceph osd set nodeep-scrub
+  delegate_to: "{{ groups['ceph_monitors'][0] }}"
+  when: inventory_hostname == groups['ceph_osds'][0]
+
+- name: restart ceph osds
+  service: name=ceph-osd-all state=restarted
```

Comment:
Are you sure that `noout`, `noscrub`, `nodeep-scrub` have no config file entry that would override the behavior you're setting when the service restarts?